### PR TITLE
Allow same name + Multiple test case fixes

### PR DESCRIPTION
**Pull request by brian16600**
Add functionality of allowing same name.
Change error message in AddCommand.java and EditCommand.java to show duplicate username, phone or email
Fix multiple test cases, removed Address Field from many test cases.

Unclear on AddressBookTest.java test, so I commented it out. Maybe someone can take a look at it?

### CODE REVIEW

**Review by junrong98 — COMMENTED** [review]
LGTM.

File: src/main/java/seedu/address/model/person/GitUsername.java [hunk 1]
```diff
@@ -5,9 +5,10 @@
  */
 public class GitUsername {
 
-    public static final String MESSAGE_CONSTRAINTS = "Github usernames should only contain alphanumeric characters.\n";
+    public static final String GIT_USERNAME_REGEX = "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+){0,2}$";
 
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
+    public static final String MESSAGE_CONSTRAINTS = "Github usernames should only contain alphanumeric"
+            + "characters or hyphens.\n";
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/person/GitUsername.java:11]: May need to consider a space here

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -43,17 +46,17 @@
...
             + PREFIX_EMAIL + "johndoe@example.com "
             + PREFIX_GIT_USERNAME + "john123";
 
     public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
-    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person with these exact same parameters already "
-            + "exists in the address book.";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.\n"
+            + MESSAGE_USAGE;
+    public static final String MESSAGE_DUPLICATE_PERSON = "At least one field of this contact must be edited! \n%1$s";
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:59]: The message does not seems to match with the variable name🤔 (ps. can ignore if is intended)

File: src/main/java/seedu/address/model/person/GitUsername.java (hunk 1 above)
Comment (brian16600) [reply src/main/java/seedu/address/model/person/GitUsername.java:11]: Fixed

**Review by snss231 — COMMENTED** [review]
Other than what JJ mentioned, LGTM

**Comment by brian16600** [comment]
Merging this PR

**Pull request merged**